src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java

```diff
@@ -1,128 +1,193 @@
 import com.minecolonies.coremod.client.gui.WindowHutGuardTower;
-import com.minecolonies.coremod.colony.CitizenData;
-import com.minecolonies.coremod.colony.Colony;
-import com.minecolonies.coremod.colony.ColonyView;
+import com.minecolonies.coremod.colony.*;
+import com.minecolonies.coremod.colony.buildings.views.MobEntryView;
 import com.minecolonies.coremod.colony.jobs.AbstractJob;
-import com.minecolonies.coremod.colony.jobs.JobGuard;
+import com.minecolonies.coremod.colony.jobs.AbstractJobGuard;
+import com.minecolonies.coremod.colony.jobs.JobKnight;
+import com.minecolonies.coremod.colony.jobs.JobRanger;
+import com.minecolonies.coremod.entity.EntityCitizen;
+import com.minecolonies.coremod.network.messages.GuardMobAttackListMessage;
 import io.netty.buffer.ByteBuf;
 import net.minecraft.entity.SharedMonsterAttributes;
+import net.minecraft.entity.monster.EntityMob;
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.inventory.EntityEquipmentSlot;
 import net.minecraft.item.ItemArmor;
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.nbt.NBTTagList;
 import net.minecraft.nbt.NBTUtil;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.Tuple;
 import net.minecraft.util.math.BlockPos;
 import net.minecraftforge.common.util.Constants;
+import net.minecraftforge.fml.common.registry.EntityEntry;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 
-import static com.minecolonies.api.util.constant.ColonyConstants.NUM_ACHIEVEMENT_FIRST;
 import static com.minecolonies.api.util.constant.ToolLevelConstants.TOOL_LEVEL_WOOD_OR_GOLD;
 
 /**
- * Building class of the guard tower.
+ * Abstract class for Guard huts.
  */
+@SuppressWarnings({"squid:MaximumInheritanceDepth", "squid:S1448"})
 public abstract class AbstractBuildingGuards extends AbstractBuildingWorker
 {
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+    private static final String NBT_TASK           = "TASK";
+    private static final String NBT_JOB            = "job";
+    private static final String NBT_ASSIGN         = "assign";
+    private static final String NBT_RETRIEVE       = "retrieve";
+    private static final String NBT_PATROL         = "patrol";
+    private static final String NBT_PATROL_TARGETS = "patrol targets";
+    private static final String NBT_TARGET         = "target";
+    private static final String NBT_GUARD          = "guard";
+    private static final String NBT_MOBS           = "mobs";
+    private static final String NBT_MOB_VIEW       = "mobview";
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+
+    ////// --------------------------- GuardTask Enum --------------------------- \\\\\\
+    public enum GuardTask
```

Comment:
I think we can put those two enums in extra files. I'd even say in the API folder.

Reply:
Hmmm. i can't move the GuardJob one because the `api` module can't reference the `main` module. and i need that for the `getGuardJob` method inside (can't use `AbstractJobGuard`, `JobRanger` or `JobKnight`

Reply:
but the GuardTask has been moved